libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -771,6 +771,53 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 	{
 		break;
 	}
+	case FunctionType::Kind::Creation:
+	{
+		solAssert(!functionType->gasSet(), "Gas limit set for contract creation.");
+		solAssert(
+			functionType->returnParameterTypes().size() == 1,
+			"Constructor should return only one type"
+		);
+
+		TypePointers argumentTypes;
+		string constructorParams;
+		for (ASTPointer<Expression const> const& arg: arguments)
+		{
+			argumentTypes.push_back(arg->annotation().type);
+			constructorParams += ", " + IRVariable{*arg}.commaSeparatedList();
+		}
+
+		ContractDefinition const* contract =
+			&dynamic_cast<ContractType const&>(*functionType->returnParameterTypes().front()).contractDefinition();
+		m_context.objects().insert(contract);
+
+		Whiskers t(R"(
+			let <memPos> := <allocateUnbounded>(datasize("<object>"))
+			datacopy(<memPos>, dataoffset("<object>"), datasize("<object>"))
+			let	<memEnd> := <abiEncode>(add(<memPos>, datasize("<object>"))<constructorParams>)
+			<?saltSet>
+				let <retVars> := <?#salt>create2<!#salt>create</#salt>(<value>, <memPos>, sub(<memEnd>, <memPos>)<?#salt>, <salt></#salt>)
```

Comment:
Ah sorry, this was just an idea! We don't have the feature yet in whiskers :)

Reply:
Ah, I was thinking it was already implemented and clicked commit suggestion